File: Q1/ans.java
import java.util.HashMap;
import java.util.Map;

public class CharacterCount {
    public static void main(String[] args) {
        String input = "mohammad";
        Map<Character, Integer> counts = new HashMap<>();

        for (char c : input.toCharArray()) {
            counts.put(c, counts.getOrDefault(c, 0) + 1);
        }

        for (Map.Entry<Character, Integer> entry : counts.entrySet()) {
            System.out.println(entry.getKey() + ": " + entry.getValue());
        }
    }
}

File: Q1/ans.js
function countCharacters(input) {
    const counts = {};
    for (const char of input) {
        counts[char] = (counts[char] || 0) + 1;
    }
    for (const [char, count] of Object.entries(counts)) {
        console.log(`${char}: ${count}`);
    }
}


countCharacters("mohammad");
